docs/build/voting-dapp-tutorial.md

```diff
@@ -0,0 +1,428 @@
+$ pact
+pact> (load "vote.repl")
+```
+And you should see an output similar to the one below (the hash will be different):
+```clojure
+"Loading vote.repl..."
+"Begin Tx 0"
+"Setting transaction keys"
+"Setting transaction data"
+"Loading vote.pact..."
+"Keyset defined"
+"Loaded module simple-vote, hash jOBJiSEH5HVsAmLNSdHq_D3Kl6TkFpCJcEWPtJD86Gc"
+"Voted A!"
+"Commit Tx 0"
+```
+
+Let's continue by adding a bit more logic. Copy this snippet just below the module declaration:
+
+```clojure
+  (defschema votes-schema
+    "Votes table schema"
+    option:string
+    count:integer)
+
+  (deftable votes:{votes-schema})
+  ```
+In the snippet we've got 2 new declaration:
+* `defschema votes-schema` -> defines the schema for our table where we will keep track of the number of votes
+* `deftable votes:{votes-schema}` -> defines the `votes` table that will use the schema defined above
+
+Now we need to update our `vote` function:
+
+```clojure
+(defun vote (key:string)
+    "Submit a new vote"
+    (enforce (contains key ["A" "B"]) "You can only vote for A or B")
+    (with-read votes key { "count" := count }
+      (update votes key { "count": (+ count 1) })
+    )
+    (format "Voted {}!" [key])
+  )
+```
+
+Again, quick recap of what's happening above:
+* we use `enforce` to ensure that only the given list of vote keys are allowed
+* we use `with-read` to first read the row data for the given key followed by an `update` call to increment the `count` column by 1
+
+We're missing a way to read the current number of votes so here it is:
+
+```clojure
+(defun getVotes:integer (key:string)
+    "Get the votes count by key"
+    (at 'count (read votes key ['count]))
+)
+```
+
+And we also need a way to initialize our table:
+
+```clojure
+(defun init ()
+    "Initialize the rows in votes table"
+    (insert votes "A" { "option": "A", "count" : 0 })
+    (insert votes "B" { "option": "B", "count" : 0 })
+)
+```
+
+If you followed the steps correctly, your code should look similar to this:
+```clojure
+(define-keyset 'vote-admin-keyset)
+
+(module simple-vote 'vote-admin-keyset
+  "Simple voting module"
+
+  (defschema votes-schema
+    "Votes table schema"
+    option:string
+    count:integer)
+
+  (deftable votes:{votes-schema})
+
+  (defun vote (key:string)
+    "Submit a new vote"
+    (enforce (contains key ["A" "B"]) "You can only vote for A or B")
+    (with-read votes key { "count" := count }
+      (update votes key { "count": (+ count 1) })
+    )
+    (format "Voted {}!" [key])
+  )
+
+  (defun getVotes:integer (key:string)
+    "Get the votes count by key"
+    (at 'count (read votes key ['count]))
+  )
+
+  (defun init ()
+    "Initialize the rows in votes table"
+    (insert votes "A" { "option": "A", "count" : 0 })
+    (insert votes "B" { "option": "B", "count" : 0 })
+  )
+)
+
+(create-table votes)
+(init)
+```
+
+Notice how we're calling `create-table` and `init` outside of our module. That's an important step, so don't miss it.
```

Comment:
Explain how code outside module executes one on run

Reply:
Added a note about this.